src/test/java/org/prebid/server/bidder/medianet/MedianetBidderTest.java

```diff
@@ -38,59 +45,137 @@ public void setup() {
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(givenBidRequest(impBuilder -> impBuilder
+                        .id("123")
+                        .video(Video.builder().build())),
+                mapper.writeValueAsString(givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        // when
+        final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .contains(BidderBid.of(Bid.builder().impid("123").build(), video, "USD"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnxNativeBidIfxNativeIsPresentInRequestImp() throws JsonProcessingException {
+        // given
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(givenBidRequest(impBuilder -> impBuilder
+                        .id("123")
+                        .xNative(Native.builder().build())),
+                mapper.writeValueAsString(givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
+
+        // when
+        final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
+
+        // then
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue()).hasSize(1)
+                .containsExactly(BidderBid.of(Bid.builder().impid("123").build(), xNative, "USD"));
+    }
+
+    @Test
+    public void makeBidsShouldReturnErrorPreBidExceptionIfNotFoundAnyImpressions() throws JsonProcessingException {
+        // given
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(givenBidRequest(impBuilder -> impBuilder
+                        .id("123")
+                        .audio(Audio.builder().build())),
+                mapper.writeValueAsString(givenBidResponse(bidBuilder -> bidBuilder.impid("123"))));
 
         // when
         final Result<List<BidderBid>> result = medianetBidder.makeBids(httpCall, null);
 
         // then
+        assertThat(result.getValue()).isEmpty();
         assertThat(result.getErrors()).hasSize(1)
-                .allMatch(error -> error.getType() == BidderError.Type.bad_server_response
-                    && error.getMessage().startsWith("Failed to decode: Unrecognized token"));
+                .allSatisfy(error -> {
+                    assertThat(error.getType()).isEqualTo(bad_server_response);
+                    assertThat(error.getMessage()).startsWith("Failed to find impression");
+                });
     }
 
     @Test
-    public void makeBidsShouldReturnEmptyListIfBidResponseIsNull() throws JsonProcessingException {
+    public void makeBitsShouldReturnEmptyValueIfSeatBitNotPresent() throws JsonProcessingException {
         // given
-        final HttpCall<BidRequest> httpCall;
-        httpCall = sampleHttpCall(givenBidRequest(), mapper.writeValueAsString(null));
+        final HttpCall<BidRequest> httpCall = sampleHttpCall(givenBidRequest(impBuilder -> impBuilder
```

Comment:
Pls, consider breaking the line after the call to `sampleHttpCall(`, also check for similar occurences.